pkg/kinds/general.go

```diff
@@ -147,17 +149,21 @@ func (m *GrafanaResourceMetadata) SetOriginInfo(info *ResourceOriginInfo) {
 }
 
 // GetOriginInfo returns the origin info stored in k8s metadata annotations
-func (m *GrafanaResourceMetadata) GetOriginInfo() *ResourceOriginInfo {
+func (m *GrafanaResourceMetadata) GetOriginInfo() (*ResourceOriginInfo, error) {
 	v, ok := m.Annotations[annoKeyOriginName]
 	if !ok {
-		return nil
+		return nil, nil
+	}
+	t, err := m.GetOriginTimestamp()
+	if err != nil {
+		return nil, err
 	}
 	return &ResourceOriginInfo{
 		Name:      v,
 		Path:      m.GetOriginPath(),
 		Key:       m.GetOriginKey(),
-		Timestamp: m.GetOriginTimestamp(),
-	}
+		Timestamp: t,
+	}, nil
```

Comment:
```suggestion
	}, err
```

we can return the part that worked and the error rather than fully blocking